tests/Io/QueryTest.php

```diff
@@ -27,6 +27,45 @@ public function testBindParams()
         */
     }
 
+    public function testNamedParams()
+    {
+        $query = new Query('select * from test where id = :id and name = :name');
+        $sql   = $query->bindParamsFromArray([
+            'id' => 100,
+            'name' => 'test'
+        ])->getSql();
+        $this->assertEquals("select * from test where id = 100 and name = 'test'", $sql);
+
+        $query = new Query('select * from test where id = :id and name = :name');
+        $sql   = $query->bindParamsFromArray([
+            ':id' => 100,
+            ':name' => 'test'
+        ])->getSql();
+        $this->assertEquals("select * from test where id = 100 and name = 'test'", $sql);
+
+        $query = new Query('select * from test where id in (:in) and name = :name');
+        $sql   = $query->bindParamsFromArray([
+            'in' => [1, 2],
+            'name' => 'test'
+        ])->getSql();
+        $this->assertEquals("select * from test where id in (1,2) and name = 'test'", $sql);
+
+        // mixed named & ?
+        $query = new Query('select * from test where id in (?) and name = :name');
+        $sql   = $query->bindParamsFromArray([
+            [1, 2],
+            'name' => 'test'
+        ])->getSql();
+        $this->assertEquals("select * from test where id in (1,2) and name = 'test'", $sql);
+
+        $query = new Query('select * from test where id in (:in) and name = ?');
+        $sql   = $query->bindParamsFromArray([
+            'in' => [1, 2],
+            'test'
+        ])->getSql();
+        $this->assertEquals("select * from test where id in (1,2) and name = 'test'", $sql);
+    }
```

Comment:
I think it's a good idea to split these into several single tests and give them a suitable name. If something should break while adding new features/changes in the future we can better identify the problem this way. What do you think?

Reply:
This is done! 😊